app/services/invoices/payments/cashfree_service.rb

```diff
@@ -0,0 +1,202 @@
+# frozen_string_literal: true
+
+module Invoices
+  module Payments
+    class CashfreeService < BaseService
+      include Customers::PaymentProviderFinder
+
+      PENDING_STATUSES = %w[PARTIALLY_PAID].freeze
+      SUCCESS_STATUSES = %w[PAID].freeze
+      FAILED_STATUSES = %w[EXPIRED CANCELLED].freeze
+
+      def initialize(invoice = nil)
+        @invoice = invoice
+
+        super(nil)
+      end
+
+      def create
+        Rails.logger.debug "invoice.total_amount_cents 0: #{invoice.total_amount_cents}"
+
+        result.invoice = invoice
+        return result unless should_process_payment?
+
+        Rails.logger.debug "invoice.total_amount_cents 1: #{invoice.total_amount_cents.positive?}"
+
+        unless invoice.total_amount_cents.positive?
+          update_invoice_payment_status(payment_status: :succeeded)
+          return result
+        end
+
+        Rails.logger.debug "invoice.total_amount_cents 2: #{invoice.total_amount_cents}"
+
+        increment_payment_attempts
+
```

Comment:
With other payment provider, this service emits/creates the payment attempt via the service provider API. It looks like this one is only creating a `Payment` resource in Lago database. Is it something that is missing or does it behave differently?

Reply:
I referred to `GoCardless` and `Adyen` providers, they create the `Payment` in the `create` method the same way and then update the status with the `update_payment_status` when the webhook is captured. Am I misunderstanding something?

Reply:
For example, in the Gocardless service, we are first emitting a payment on the service (https://github.com/getlago/lago-api/blob/main/app/services/invoices/payments/gocardless_service.rb#L43C29-L43C54) and then creating the payment inside Lago's database with `provider_payment_id: gocardless_result.id`. In this case `gocardless_result.id` is the ID of the payment inside gocardless database

Reply:
As I mentioned earlier, I'm working on implementing the `Payment Link` feature for `Cashfree Payments`. Therefore, the payment emission process for Cashfree does not apply in this scenario. I create the `Payment` when the `create` event occurs and then update the status when I receive a webhook. Given this scenario, what do you think I should do?

Reply:
Ok, thank you for the clarification @AyushChothe.
In that case I guess the current approach should work. Could you maybe add some comments in the code to explain the workflow?